lib/big_brother/cluster.rb

```diff
@@ -237,27 +236,31 @@ def down_file_exists?
       @down_file.exists?
     end
 
-    def incorporate_state(cluster)
+    def incorporate_state(original_cluster)
       ipvs_state = BigBrother.ipvs.running_configuration
       if ipvs_state[fwmark.to_s] && ipvs_state[_relay_fwmark.to_s].nil?
         BigBrother.logger.info "Adding new remote relay cluster #{to_s}"
         BigBrother.ipvs.start_cluster(_relay_fwmark, @scheduler)
       end
 
-      if ipvs_state[fwmark.to_s] && ipvs_state.fetch(_relay_fwmark.to_s, []).empty?
-        _active_nodes.each do |node|
-          actual_node = cluster.find_node(node.address, node.port)
-          BigBrother.ipvs.start_node(_relay_fwmark, actual_node.address, actual_node.weight)
+      nodes.each do |node|
+        original_node = original_cluster.find_node(node.address, node.port)
+        node.incorporate_state(original_node)
+
+        if original_node.nil? && active_active?
+          _start_node(node)
+        elsif ipvs_state[fwmark.to_s] && ipvs_state.fetch(_relay_fwmark.to_s, []).empty?
+          BigBrother.ipvs.start_node(_relay_fwmark, node.address, node.weight)
```

Comment:
This is just an extra check to ensure that the `relay_fwmark` node is started in ipvs if the local fwmark is running. This is legacy from before, and I included this in the conditional because ling 251 runs this similar code and the situations are typically exclusive. It technically saves us a conditional check, but I can see how it would be helpful to make this more readable.